dockerhub/README.md

```diff
@@ -0,0 +1,186 @@
+
+    acq_data_path = "/rapids/data/mortgage/acq"
+    perf_data_path = "/rapids/data/mortgage/perf"
+    col_names_path = "/rapids/data/mortgage/names.csv"
+    start_year = 2000
+    end_year = 2002 # end_year is not inclusive
+    part_count = 11 # the number of data files to train against
+
+These are the paths to data, the number of years on which to perform ETL, and the number of parts to use for training.
+
+Note: the entire mortgage dataset is 68 quarters, broken into 112 parts so that each part is (on average) 1.7GB. Reducing the number of parts, `part_count`, reduces how much data is input to XGBoost training. Adjusting the `start_year` and`end_year` changes how many years on which to perform ETL.
+
+* * *
+
+## Useful Tips and Tools for Monitoring and Debugging
+
+*   The Dask Dashboard. This cell from the notebook
+
+    import subprocess
+
+    cmd = "hostname --all-ip-addresses"
+    process = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE)
+    output, error = process.communicate()
+    IPADDR = str(output.decode()).split()[0]
+    _client = IPADDR + str(":8786")
+
+    client = dask.distributed.Client(_client)
+    client
+
+initializes the Dask client. Once done, you should see output like
+
+    Client
+
+    Scheduler: tcp://172.17.0.3:8786
+    Dashboard: http://172.17.0.3:8787/status
+    Cluster
+
+    Workers: 8
+    Cores: 8
+    Memory: 0 B
+
+Note: clicking `Dashboard: http://172.17.0.3:8787/status`  may not work. The IP address in the link may be the networking device in the container. You have to use the IP address of the primary networking device, which should be the one you used to access the Jupyter notebook server. `http://your.ip.address:8787/status`
+
+Once there, you can look at active processes, view the (delayed) task graph, and read through individual worker logs.
+
+*   Individual Worker Logs
+
+When you hit an unusual error, there are often traces of the error in the individual worker logs accessible from the Dashboard status page (see above). CUDA out-of-memory errors will show up here.
+
+* * *
+
+## Common Errors
+
+*   Running out of device memory
+
+    *   ETL processes may involve creating many copies of data in device memory, resulting in sporadic spikes of memory utilization
+
+        *   Memory utilization which exceeds available device resources will cause the worker to crash
+        *   Because we've disabled experimental features that would enable the worker to recover, restart its work, or share its work with other workers, the error is silently propagated forward in the Dask's delayed task graph. This can manifest itself in the form of unusually short ETL times (sub-millisecond timescale)
+        *   An error may be raised by another routine referring to `NoneType` in `data` or similar
+        *   Note: it is possible to enable these _experimental_ features by changing the Dask `export` parameters in the `/rapids/utils/dask-setup.sh`:
```

Comment:
Let's get rid of this experimental session.

I think there shouldn't be process `nanny`ing, yet. We don't support it.

Reply:
Removed.